pkg/controller/custom.go

```diff
@@ -106,16 +106,17 @@ func (c *Controller) getCustomCertificates(ctx context.Context, tenant *miniov2.
 				expiresInSeconds := int64(math.Mod(expiresIn.Seconds(), 60))
 				expiresInHuman := fmt.Sprintf("%v days, %v hours, %v minutes, %v seconds", expiresInDays, expiresInHours, expiresInMinutes, expiresInSeconds)
 
+				hoursMinutesSecondsAllZero := expiresInHours == 0 && expiresInMinutes == 0 && expiresInSeconds == 0
 				if expiresInDays >= 10 && expiresInDays < 30 {
 					c.recorder.Event(tenant, corev1.EventTypeWarning, "CertificateExpiring", fmt.Sprintf("%s certificate '%s' is expiring in %d days", certType, secret.Name, expiresInDays))
 				}
 				if expiresInDays > 0 && expiresInDays < 10 {
 					c.recorder.Event(tenant, corev1.EventTypeWarning, "CertificateExpiryImminent", fmt.Sprintf("%s certificate '%s' is expiring in %d days", certType, secret.Name, expiresInDays))
 				}
-				if expiresInDays > 0 && expiresInDays < 1 {
+				if expiresInDays == 0 && !hoursMinutesSecondsAllZero {
```

Comment:
I think this is more readable (and it doesn't require the `hoursMinutesSecondsAllZero` variable.
```suggestion
				if expiresIn > 0 && expiresIn < 24*time.Hour {
```